plugins/invisible-typing/src/Icon.tsx

```diff
@@ -17,43 +17,41 @@ export const Icon = (props: { type?: { analyticsName: string } }) => {
   const channelWise = cfg.get("button", true) ? cfg.get("channelWise", true) : false;
   const channelsList = cfg.get("channels", { [channelId]: globalInvisible });
   const [enabled, setEnabled] = React.useState(
-    channelWise ? channelsList[channelId] : globalInvisible,
+    channelWise ? channelsList[channelId] ?? globalInvisible : globalInvisible,
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Use of nullish coalescing operator improves handling of undefined channel settings.

This change ensures that if a specific channel's visibility setting is undefined, the global visibility setting is used as a fallback. It's a robust way to handle potentially missing configurations.

```suggestion
if ((channelWise ? channels[channelId] : undefined) ?? globalInvisible) {
```